### ISSUE TITLE
Story: Add XP backend

### ISSUE THREAD

**Issue by agraubert** [agraubert/Beymax#39]
I would like to extend the story feature with an xp system for users on the server. Users will start at level 1 and gain level by earning xp from the following sources:

* [ ] Using commands. This has a potential abuse, so I've listed a few potential systems below
  - [ ] Grant 1xp per command (greatest abuse potential)
  - [ ] Grant 10xp each time a command is used for the first time
  - [ ] Grant 5xp each time a command is used for the first time each week
* [ ] Being active in the server: At the end of each week, any user that posted a message gains 5xp
  - It's possible that we may want to tier this based on the number of messages posted
* [ ] Playing a game: Earn xp based on your score when you ended the game
* [ ] Getting an automated reaction from Beymax (requires the event system #34 to be merged in)

**Comment by agraubert** [comment]
I'm going for a level progression like the following:
```python
def xp_for(level):
    if level <= 2:
        return 10
    else:
        return (2*xp_for(level-1)-xp_for(level-2))+5
```
which gives the following progression for levels 2-12:
`[10, 15, 25, 40, 60, 85, 115, 150, 190, 235]`

Each number represents the additional xp to the next level, not the cumulative xp required

**Comment by agraubert** [comment]
Closed in 9764d9d

Should have done it in a separate branch, but oh well
